osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -40,10 +163,57 @@ public override bool HandleMovement(MoveSelectionEvent moveEvent)
                     continue;
                 }
 
-                h.Position += moveEvent.InstantDelta;
+                h.Position += delta;
             }
 
             return true;
         }
+
+        private Quad getSelectionQuad()
+        {
+            Vector2 minPosition = new Vector2(float.MaxValue, float.MaxValue);
+            Vector2 maxPosition = new Vector2(float.MinValue, float.MinValue);
+
+            // Go through all hitobjects to make sure they would remain in the bounds of the editor after movement, before any movement is attempted
+            foreach (var h in SelectedHitObjects.OfType<OsuHitObject>())
+            {
+                if (h is Spinner)
+                {
+                    // Spinners don't support position adjustments
+                    continue;
+                }
+
+                // Stacking is not considered
+                minPosition = Vector2.ComponentMin(minPosition, Vector2.ComponentMin(h.EndPosition, h.Position));
+                maxPosition = Vector2.ComponentMax(maxPosition, Vector2.ComponentMax(h.EndPosition, h.Position));
+            }
+
+            Vector2 size = maxPosition - minPosition;
+
+            return new Quad(minPosition.X, minPosition.Y, size.X, size.Y);
+        }
+
+        /// <summary>
+        /// Returns rotated position from a given point.
+        /// </summary>
+        /// <param name="p">The point.</param>
+        /// <param name="center">The center to rotate around.</param>
+        /// <param name="angle">The angle to rotate (in degrees).</param>
+        internal static Vector2 Rotate(Vector2 p, Vector2 center, float angle)
```

Comment:
nit again but this function has the freedom spelling of the word :smile: 
```suggestion
        internal static Vector2 Rotate(Vector2 p, Vector2 centre, float angle)
```